rust/lance-encoding/src/buffer.rs

```diff
@@ -223,6 +226,7 @@ impl LanceBuffer {
     ///
     /// If the underlying buffer is not properly aligned, this will involve a copy of the data
     pub fn borrow_to_typed_slice<T: ArrowNativeType>(&mut self) -> impl AsRef<[T]> {
+        check_little_endian();
```

Comment:
How about we just disallow building Lance when the targeted platform is not little endian?